tokio-stream/tests/stream_close.rs

```diff
@@ -0,0 +1,35 @@
+use tokio_stream::{Stream, StreamExt, StreamNotifyClose};
+
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+struct Test {
+    state: i32,
+}
```

Comment:
Your custom stream is the same as `tokio_stream::iter(vec![0,1])`.
```suggestion
struct Test {
    state: i32,
}
```